src/OrchardCore.Modules/OrchardCore.Users/Services/SetupEventHandler.cs

```diff
@@ -29,10 +29,11 @@ public SetupEventHandler(IUserService userService)
                 UserName = properties.TryGetValue(SetupConstants.AdminUsername, out var adminUserName) ? adminUserName?.ToString() : String.Empty,
                 UserId = properties.TryGetValue(SetupConstants.AdminUserId, out var adminUserId) ? adminUserId?.ToString() : String.Empty,
                 Email = properties.TryGetValue(SetupConstants.AdminEmail, out var adminEmail) ? adminEmail?.ToString() : String.Empty,
-                RoleNames = new string[] { "Administrator" },
                 EmailConfirmed = true
             };
 
+            user.RoleNames.Add("Administrator");
```

Comment:
I think it is safer to use `RoleNames = new string[] { "Administrator" },` but up to you.

Reply:
Here the little fix is to not set the `IList RoleNames` to a string array, otherwise it saves in the json document the following.

    "RoleNames":{"$type":"System.String[], System.Private.CoreLib","$values":["Administrator"]}

In place of

    "RoleNames":["Administrator"]

Reply:
Then we add a role to an empty list because the user is created just before.